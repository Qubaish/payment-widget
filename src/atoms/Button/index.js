import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledButton = styled.button`
    width: ${props => props.type === 'sm' ? '50px' : '100%'};
    background: ${props => props.type === 'sm' ? '#F5F5F5' : '#307EF7'};
    border: ${props => props.type === 'sm' ? '1px solid #999' : '1px solid #307EF7'};
    border-radius: 10px;
    height: ${props => props.type === 'lg' ? '30px' : 'auto'};
    color: ${props => props.type === 'sm' ? '#999' : '#fff'};
    cursor: pointer;
    :focus {
        outline: none;
    }
`;

function Button({text, type, onClick, style}) {
    return (
        <StyledButton
            type={type}
            onClick={onClick}
            data-testid="btn"
            aria-label={`${text} button`}
            style={style}
        >
            {text}
        </StyledButton>
    );
};

Button.prototype = {
    onClick: PropTypes.func,
    type: PropTypes.oneOf(['sm', 'lg']),
    text: PropTypes.string,
};

Button.defaultProps = {
    type: 'sm',
    text: 'close'
};

export default Button;