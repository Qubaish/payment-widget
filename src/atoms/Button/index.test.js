import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';

describe('<Button />', () => {
    it('should render a sm button', () => {
        const { getByTestId} = render(<Button type="sm" text='close'/>);
        const elem = getByTestId('btn');
        expect(elem.innerHTML).toBe('close');
    });
    it('should render a lg button', () => {
        const { getByTestId } = render(<Button type="lg" text='submit'/>);
        const elem = getByTestId('btn');
        expect(elem.innerHTML).toBe('submit');

    });
    it('should check css of lg button', () => {
        const { getByTestId } = render(<Button type="lg" text='submit'/>);
        const elem = getByTestId('btn');
        expect(elem).toHaveStyle('color: #fff');
        
    });
    it('should fire an event', () => {
        const { getByTestId, getByText } = render(<Button type="lg" text='event'/>);
        const elem = getByTestId('btn');
        fireEvent.click(getByText("event"));
        expect(elem.innerHTML).toBe('event');
    });
    it('checking snapshot', () => {
        const { asFragment } = render(<Button type="lg" text='event'/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
