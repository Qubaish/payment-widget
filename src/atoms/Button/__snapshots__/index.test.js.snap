// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> checking snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="event button"
    class="sc-AxjAm kzTiRY"
    data-testid="btn"
    type="lg"
  >
    event
  </button>
</DocumentFragment>
`;
